src/components/Menu/Menu.minors.tsx

```diff
@@ -33,10 +38,24 @@ export function Item({
   href,
   toggle = false,
   icon,
+  /** Function called if item is toggled open */
+  onOpen,
+  /** Function called if item is toggled close */
+  onClose,
   ...props
 }) {
+  const init = useRef(true);
   const [open, setOpen] = useState(!toggle);
 
+  useEffect(() => {
+    // Prevent calling onClose on mount
+    if (init.current) {
+      init.current = false;
+    } else {
+      open ? onOpen && onOpen() : onClose && onClose();
+    }
```

Comment:
Not terribly important but this might be more clearly written as:


```suggestion
    if (init.current) {
      init.current = false;
    } else if (open) {
      onOpen?.();
    } else {
      onClose?.());
    }
```